packages/core/src/renderTexture/RenderTexturePool.js

```diff
@@ -0,0 +1,177 @@
+            minWidth = nextPow2(minWidth);
+            minHeight = nextPow2(minHeight);
+            key = ((minWidth & 0xFFFF) << 16) | (minHeight & 0xFFFF);
+        }
+
+        if (!this.texturePool[key])
+        {
+            this.texturePool[key] = [];
+        }
+
+        let renderTexture = this.texturePool[key].pop();
+
+        if (!renderTexture)
+        {
+            renderTexture = this.createTexture(minWidth, minHeight);
+        }
+
+        renderTexture.filterPoolKey = key;
+        renderTexture.setResolution(resolution);
+
+        return renderTexture;
+    }
+
+    /**
+     * Gets extra texture of the same size as current renderTexture
+     *
+     * @param {number} [resolution] resolution, by default its the same as in current renderTexture
+     * @returns {PIXI.RenderTexture}
+     */
+    getFilterTexture(resolution)
+    {
+        const rt = this.renderer.renderTexture.current;
+
+        const filterTexture = this.getOptimalTexture(rt.width, rt.height, resolution || rt.baseTexture.resolution);
+
+        filterTexture.filterFrame = rt.filterFrame;
+
+        return filterTexture;
+    }
+
+    /**
+     * Place a render texture back into the pool.
+     * @param {PIXI.RenderTexture} renderTexture - The renderTexture to free
+     */
+    returnTexture(renderTexture)
+    {
+        const key = renderTexture.filterPoolKey;
+
+        renderTexture.filterFrame = null;
+        this.texturePool[key].push(renderTexture);
+    }
+
+    /**
+     * Alias for returnTexture, to be compliant with FilterSystem interface
+     * @param {PIXI.RenderTexture} renderTexture - The renderTexture to free
+     */
+    returnFilterTexture(renderTexture)
+    {
+        this.returnTexture(renderTexture);
+    }
+
+    /**
+     * Clears the pool
+     *
+     * @member {boolean} [destroyTextures=true] destroy all stored textures
+     */
+    clear(destroyTextures)
+    {
+        destroyTextures = destroyTextures !== false;
+        if (destroyTextures)
+        {
+            for (const i in this.texturePool)
+            {
+                const textures = this.texturePool[i];
+
+                if (textures)
+                {
+                    for (let j = 0; j < textures.length; j++)
+                    {
+                        textures[j].destroy(true);
+                    }
+                }
+            }
+        }
+
+        this.texturePool = {};
+    }
+
+    /**
+     * resizes all textures who has the same size as screen
+     */
+    resize()
+    {
+        const screenKey = RenderTexturePool.SCREEN_KEY;
+        const textures = this.texturePool[screenKey];
+
+        if (textures)
+        {
+            for (let j = 0; j < textures.length; j++)
+            {
+                textures[j].destroy(true);
+            }
+        }
+        this.texturePool[screenKey] = [];
+
+        this._pixelsWidth = this.renderer.view.width;
+        this._pixelsHeight = this.renderer.view.height;
+    }
+}
+
```

Comment:
what is Screen Key used for? Mind adding some docs for it?

Reply:
ok